tests/e2e/deployment-test.sh

```diff
@@ -68,6 +68,16 @@ fi
 
 sudo mv linux-amd64/helm /usr/local/bin/helm
 
+# Create E2E Sample App
+gcloud functions deploy sample_app \
+--runtime python37 \
+--trigger-http \
+--source sample_app/ \
+--allow-unauthenticated
```

Comment:
With this Cloud Function allowing unauthenticated requests, we are telegraphing open endpoint for anyone to use. Some malicious actor could flood it with requests and make us a costly GCP bill.

Reply:
I'd suggest using this path https://cloud.google.com/functions/docs/securing/authenticating#authenticating_developer_testing to authenticate the requests.